sdk/src/main/scala/com/horizen/account/state/WithdrawalMsgProcessor.scala

```diff
@@ -0,0 +1,195 @@
+  val gasSpentForAddNewWithdrawalReqCmd: java.math.BigInteger = java.math.BigInteger.ONE
+  val gasSpentForAddNewWithdrawalReqFailure: java.math.BigInteger = java.math.BigInteger.ONE
+  val gasSpentForGenericFailure: java.math.BigInteger = java.math.BigInteger.ONE
+
+  val MAX_WITHDRAWAL_REQS_NUM_PER_EPOCH = 3900
+  val DUST_THRESHOLD_IN_WEI: java.math.BigInteger = ZenWeiConverter.convertZenniesToWei(ZenCoinsUtils.getMinDustThreshold(ZenCoinsUtils.MC_DEFAULT_FEE_RATE))
+
+
+  override def canProcess(msg: Message, view: AccountStateView): Boolean = {
+    fakeSmartContractAddress.equals(msg.getTo)
+  }
+
+  override def process(msg: Message, view: AccountStateView): ExecutionResult = {
+    //TODO: check errors in Ethereum, maybe for some kind of errors there a predefined types or codes
+
+    try {
+      if (!canProcess(msg, view)) {
+        new InvalidMessage(new IllegalArgumentException(s"Cannot process message $msg"))
+      }
+      else {
+
+        val functionSig = BytesUtils.toHexString(getOpCodeFromData(msg.getData))
+
+        if (getListOfWithdrawalReqsCmdSig.equals(functionSig)) {
+          getListOfWithdrawalReqRecords(msg, view) match {
+            case Success(result) =>
+              result
+            //TODO log
+            case Failure(exception: IllegalArgumentException) =>
+              new ExecutionFailed(gasSpentForGetListOfWithdrawalReqsFailure, exception)
+            case Failure(exception) =>
+              log.error(s"Error while requesting Withdrawal Request list: ${exception.getMessage}", exception)
+              new ExecutionFailed(gasSpentForGetListOfWithdrawalReqsFailure, new Exception(exception))
+          }
+        }
+        else if (addNewWithdrawalReqCmdSig.equals(functionSig)) {
+          addWithdrawalRequest(msg, view) match {
+            case Success(result) =>
+              result
+            //TODO log
+            case Failure(exception: IllegalArgumentException) =>
+              new ExecutionFailed(gasSpentForAddNewWithdrawalReqFailure, exception)
+            case Failure(exception) =>
+              log.error(s"Error while adding a new Withdrawal Request: ${exception.getMessage}", exception)
+              new ExecutionFailed(gasSpentForAddNewWithdrawalReqFailure, new Exception(exception))
+          }
+        }
+        else {
+          log.error(s"Requested function does not exist. Function signature: $functionSig")
+          new ExecutionFailed(gasSpentForGenericFailure, new IllegalArgumentException(s"Requested function does not exist"))
+        }
+      }
+    }
+    catch {
+      case e: Exception =>
+        log.error(s"Exception while processing message: $msg", e)
+        new ExecutionFailed(gasSpentForGenericFailure, e)
+    }
+
+  }
+
+  private def getWithdrawalEpochCounter(view: AccountStateView, epochNum: Int) = {
+    val key = getWithdrawalEpochCounterKey(epochNum)
+    val wrCounterInBytes = view.getAccountStorageBytes(fakeSmartContractAddress.address(), key).get
```

Comment:
knowing that the value here is and integer (4 bytes), we may use the RAW strategy to read&write.
Like int Eth, they prepend the value with zero bytes to reach the 32 bytes size.
This will save 1 slot.